Converting `terraform-aws-github-runner`: panic: `must not call MapVal with empty map`

Using the latest bits, I'm getting the following panic converting https://github.com/philips-labs/terraform-aws-github-runner:

```
panic: must not call MapVal with empty map

goroutine 56 [running]:

github.com/zclconf/go-cty/cty.MapVal(0x1065bc450?)

	/Users/justin/go/pkg/mod/github.com/zclconf/go-cty@v1.13.1/cty/value_init.go:211 +0x464

github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/convert.camelCaseObjectAttributes({{{0x1065bc450?, 0x14000b51a80?}}, {0x10624c820?, 0x140007b3950?}})

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.1-0.20230427084552-7c063858d2f5/pkg/tf2pulumi/convert/tf.go:1475 +0x530

github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/convert.convertVariable(0x1062460a0?, 0x140022a0180?, 0x140009b4820)

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.1-0.20230427084552-7c063858d2f5/pkg/tf2pulumi/convert/tf.go:1505 +0x234

github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/convert.translateModuleSourceCode(0x106241b40?, {0x1065c4e40, 0x107595dc8}, {0x1400049d040, 0x4d}, {0x1065c4dc0?, 0x140005f64e0?}, {0x140009473d0, 0xc}, {0x1065a0cc0, ...})

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.1-0.20230427084552-7c063858d2f5/pkg/tf2pulumi/convert/tf.go:2443 +0x14e4

github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/convert.translateModuleSourceCode(0x140003cef30?, {0x1065c4e40, 0x107595dc8}, {0x1400035a280, 0x41}, {0x1065c4dc0?, 0x140005f64e0?}, {0x105ff32d0, 0x1}, {0x1065a0cc0, ...})

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.1-0.20230427084552-7c063858d2f5/pkg/tf2pulumi/convert/tf.go:2263 +0x3c6c

github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tf2pulumi/convert.TranslateModule(...)

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.43.1-0.20230427084552-7c063858d2f5/pkg/tf2pulumi/convert/tf.go:2523

main.(*tfConverter).ConvertProgram(0x14000ab17b8, {0x104e515fc?, 0x40?}, 0x1400050e420)

	/Users/justin/go/src/github.com/pulumi/pulumi-converter-terraform/cmd/pulumi-converter-terraform/main.go:53 +0x1c8

github.com/pulumi/pulumi/sdk/v3/proto/go._Converter_ConvertProgram_Handler.func1({0x1065b9350, 0x140005f4d20}, {0x10643b260?, 0x1400050e420})

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.65.1/proto/go/converter_grpc.pb.go:122 +0x74

github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x1065b9350, 0x140005f49c0}, {0x10643b260, 0x1400050e420}, 0x140005f61e0, 0x140005e8390)

	/Users/justin/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x308

github.com/pulumi/pulumi/sdk/v3/proto/go._Converter_ConvertProgram_Handler({0x1062ec9e0?, 0x107595dc8}, {0x1065b9350, 0x140005f49c0}, 0x140008d81c0, 0x140009480e0)

	/Users/justin/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.65.1/proto/go/converter_grpc.pb.go:124 +0x138

google.golang.org/grpc.(*Server).processUnaryRPC(0x14000792780, {0x1065c2e00, 0x14000583380}, 0x14000ac0120, 0x140008a7650, 0x10751e8f8, 0x0)

	/Users/justin/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:1345 +0xc90

google.golang.org/grpc.(*Server).handleStream(0x14000792780, {0x1065c2e00, 0x14000583380}, 0x14000ac0120, 0x0)

	/Users/justin/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:1722 +0x82c

google.golang.org/grpc.(*Server).serveStreams.func1.2()

	/Users/justin/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:966 +0x84

created by google.golang.org/grpc.(*Server).serveStreams.func1

	/Users/justin/go/pkg/mod/google.golang.org/grpc@v1.54.0/server.go:964 +0x28c
```

https://github.com/pulumi/pulumi-terraform-bridge/blob/7c063858d2f53691d4f341274fadbd7f6b7c7391/pkg/tf2pulumi/convert/tf.go#L1475

`cty.MapVal` will panic on empty maps. So will `cty.ListVal`.

--- a/pkg/tf2pulumi/convert/testdata/object_config_rename/experimental_pcl/main.pp
+++ b/pkg/tf2pulumi/convert/testdata/object_config_rename/experimental_pcl/main.pp
@@ -10,6 +10,9 @@ config "objectListConfig" "list(object({firstMember=number, secondMember=string}
     secondMember = "hello"
   }]
 }
+config "objectListConfigEmpty" "list(object({firstMember=number, secondMember=string}))" {
+  default = []
+}
 config "objectMapConfig" "map(object({firstMember=number, secondMember=string}))" {
   default = {
     hello = {
@@ -17,6 +20,9 @@ config "objectMapConfig" "map(object({firstMember=number, secondMember=string}))
       secondMember = "hello"
     }
   }
+}
+config "objectMapConfigEmpty" "map(object({firstMember=number, secondMember=string}))" {
+  default = {}
 }
 resource "usingSimpleObjectConfig" "simple:index:resource" {
   inputOne = simpleObjectConfig.firstMember

--- a/pkg/tf2pulumi/convert/testdata/object_config_rename/main.tf
+++ b/pkg/tf2pulumi/convert/testdata/object_config_rename/main.tf
@@ -24,6 +24,15 @@ variable "object_list_config" {
     }]
 }
 
+variable "object_list_config_empty" {
+    type = list(object({
+        first_member = number,
+        second_member = string
+    }))
+
+    default = []
+}
+
 variable "object_map_config" {
     type = map(object({
         first_member = number,
@@ -36,6 +45,15 @@ variable "object_map_config" {
              second_member = "hello"
         }
     }
+}
+
+variable "object_map_config_empty" {
+    type = map(object({
+        first_member = number,
+        second_member = string
+    }))
+
+    default = {}
 }
 
 resource "simple_resource" "using_simple_object_config" {

--- a/pkg/tf2pulumi/convert/tf.go
+++ b/pkg/tf2pulumi/convert/tf.go
@@ -1425,8 +1425,13 @@ func convertBody(sources map[string][]byte, scopes *scopes, fullyQualifiedPath s
 // camelCaseObjectAttributes rewrites the attributes of objects to camelCase and returns the modified value.
 // when the input is a list of objects or map of objects, those are modified recursively.
 func camelCaseObjectAttributes(value cty.Value) cty.Value {
+	// if the value is null, return as is
+	if value.IsNull() {
+		return value
+	}
+
 	// handle type object({...})
-	if value.Type().IsObjectType() {
+	if value.Type().IsObjectType() && value.LengthInt() > 0 {
 		modifiedAttributes := map[string]cty.Value{}
 		for propertyKey, propertyValue := range value.AsValueMap() {
 			modifiedValue := camelCaseObjectAttributes(propertyValue)
@@ -1436,7 +1441,7 @@ func camelCaseObjectAttributes(value cty.Value) cty.Value {
 	}
 
 	// handle type list(...)
-	if value.Type().IsListType() {
+	if value.Type().IsListType() && value.LengthInt() > 0 {
 		modifiedValues := make([]cty.Value, value.LengthInt())
 		for index, element := range value.AsValueSlice() {
 			modifiedValues[index] = camelCaseObjectAttributes(element)
@@ -1447,7 +1452,7 @@ func camelCaseObjectAttributes(value cty.Value) cty.Value {
 
 	// handle set(...) and convert it to list(...)
 	// because we simplify sets to lists
-	if value.Type().IsSetType() {
+	if value.Type().IsSetType() && value.LengthInt() > 0 {
 		modifiedValues := make([]cty.Value, value.LengthInt())
 		for index, element := range value.AsValueSet().Values() {
 			modifiedValues[index] = camelCaseObjectAttributes(element)
@@ -1456,7 +1461,7 @@ func camelCaseObjectAttributes(value cty.Value) cty.Value {
 		return cty.ListVal(modifiedValues)
 	}
 
-	if value.Type().IsTupleType() {
+	if value.Type().IsTupleType() && value.LengthInt() > 0 {
 		tupleValues := make([]cty.Value, value.LengthInt())
 		for index, element := range value.AsValueSlice() {
 			tupleValues[index] = camelCaseObjectAttributes(element)
@@ -1466,7 +1471,7 @@ func camelCaseObjectAttributes(value cty.Value) cty.Value {
 	}
 
 	// handle type map(object({...}))
-	if value.Type().IsMapType() {
+	if value.Type().IsMapType() && value.LengthInt() > 0 {
 		modifiedAttributes := map[string]cty.Value{}
 		for propertyKey, propertyValue := range value.AsValueMap() {
 			modifiedValue := camelCaseObjectAttributes(propertyValue)

